pkg/sql/logictest/tests/5node-disk/5node-disk_test: TestLogic_distsql_enum failed

pkg/sql/logictest/tests/5node-disk/5node-disk_test.TestLogic_distsql_enum [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/8120704?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/8120704?buildTab=artifacts#/) on master @ [6723e00a46aaa3ea575093bd82a02b7d6f6b131b](https://github.com/cockroachdb/cockroach/commits/6723e00a46aaa3ea575093bd82a02b7d6f6b131b):


```
=== RUN   TestLogic_distsql_enum
    test_log_scope.go:161: test logs captured to: /artifacts/tmp/_tmp/6e9d563e9f7703abdf07517fb35efb2e/logTestLogic_distsql_enum64606162
    test_log_scope.go:79: use -show-logs to present logs inline
[10:56:54] setting distsql_workmem='48911B';
    logic.go:2872: 
         
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/2658/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/sql/logictest/tests/5node-disk/5node-disk_test_/5node-disk_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/distsql_enum:47: SELECT start_key, end_key, replicas, lease_holder from [SHOW RANGES FROM TABLE t1 WITH DETAILS] ORDER BY lease_holder
        expected:
            start_key           end_key                 replicas  lease_holder
            <before:/Table/53>  …/1/0                   {1}       1
            …/1/0               …/1/10                  {1}       1
            …/1/10              …/1/20                  {2}       2
            …/1/20              <after:/Table/110/1/0>  {3}       3
            
        but found (query options: "colnames") :
            start_key           end_key                 replicas  lease_holder
            …/1/0               …/1/10                  {1}       1
            <before:/Table/53>  …/1/0                   {1}       1
            …/1/10              …/1/20                  {2}       2
            …/1/20              <after:/Table/110/1/0>  {3}       3
[10:56:55] --- done: /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/2658/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/sql/logictest/tests/5node-disk/5node-disk_test_/5node-disk_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/distsql_enum with config 5node-disk: 6 tests, 1 failures
    logic.go:3990: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/2658/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/sql/logictest/tests/5node-disk/5node-disk_test_/5node-disk_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/distsql_enum:56: error while processing
    logic.go:3990: /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/2658/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/sql/logictest/tests/5node-disk/5node-disk_test_/5node-disk_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/distsql_enum:56: too many errors encountered, skipping the rest of the input
    panic.go:522: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/6e9d563e9f7703abdf07517fb35efb2e/logTestLogic_distsql_enum64606162
--- FAIL: TestLogic_distsql_enum (2.33s)
```
<p>Parameters: <code>TAGS=bazel,gss</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/sql-queries
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestLogic_distsql_enum.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-22852

--- a/pkg/sql/logictest/testdata/logic_test/distsql_enum
+++ b/pkg/sql/logictest/testdata/logic_test/distsql_enum
@@ -45,7 +45,7 @@ ALTER TABLE t1 EXPERIMENTAL_RELOCATE VALUES (ARRAY[1], 0), (ARRAY[2], 10), (ARRA
 ALTER TABLE t2 EXPERIMENTAL_RELOCATE VALUES (ARRAY[1], 0), (ARRAY[2], 10), (ARRAY[3], 20);
 
 query TTTI colnames
-SELECT start_key, end_key, replicas, lease_holder from [SHOW RANGES FROM TABLE t1 WITH DETAILS] ORDER BY lease_holder
+SELECT start_key, end_key, replicas, lease_holder from [SHOW RANGES FROM TABLE t1 WITH DETAILS] ORDER BY lease_holder, start_key
 ----
 start_key           end_key                 replicas  lease_holder
 <before:/Table/53>  …/1/0                   {1}       1
@@ -54,7 +54,7 @@ start_key           end_key                 replicas  lease_holder
 …/1/20              <after:/Table/110/1/0>  {3}       3
 
 query TTTI colnames
-SELECT start_key, end_key, replicas, lease_holder from [SHOW RANGES FROM TABLE t2 WITH DETAILS] ORDER BY lease_holder
+SELECT start_key, end_key, replicas, lease_holder from [SHOW RANGES FROM TABLE t2 WITH DETAILS] ORDER BY lease_holder, start_key
 ----
 start_key                 end_key       replicas  lease_holder
 …/1/0                     …/1/10        {1}       1

